src/shogun/solver/LDASolver.h

```diff
@@ -164,12 +165,14 @@ namespace shogun
 			centered_class[i] = SGMatrix<T>(num_features, m_class_count[i]);
 			linalg::zero(centered_class[i]);
 		}
-		for (index_t i = 0; i < num_vectors; ++i)
+		counter=0;
```

Comment:
need to track the row number because the matrix iterator is along elements rather than the column vectors